### PR TITLE
Set derivative extrapolation to zeros

### LINKED ISSUE #10195: Derivative of spline with 'const' extrapolation is also extrapolted with 'const' instead of zero
Please label as: scipy.interpolate (I can't figure out how to do it...)

When using scipy.interpolate.InterpolatedUnivariateSpline with option ext='const' to extrapolate values with the boundry value, the derivative also is extrapolated with its boundry values, instead of zero. Please see figure below:

Figure_1

### Reproducing code example:

```
import numpy as np
import matplotlib.pyplot as plt
from scipy.interpolate import InterpolatedUnivariateSpline

x_values = [1, 2, 4, 6, 8.5]
y_values = [0.5, 0.8, 1.3, 2.5, 5]

f = InterpolatedUnivariateSpline(x_values, y_values, ext='const', k=3)

x = np.linspace(0, 10, 1000)
y = f(x)

plt.scatter(x_values, y_values)
plt.plot(x, y, label = 'f(x)')
plt.plot(x, f.derivative()(x), label = 'df(x)/dx')
plt.plot(x[1:], np.diff(y)/np.diff(x), label = 'actual df(x)/dx')
plt.legend()
plt.grid()
plt.show()

```

### Scipy/Numpy/Python version information:
('1.2.1', '1.16.2', sys.version_info(major=2, minor=7, micro=16, releaselevel='final', serial=0))

### PR DESCRIPTION
**Pull request by bharatr21**
Resolve #10195 by setting the extrapolation of the derivative to `'zeros'` instead of `'const'` in `fitpack2.py`

### CODE REVIEW

File: scipy/interpolate/fitpack2.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ def derivative(self, n=1):
 
         """
         tck = fitpack.splder(self._eval_args, n)
-        return UnivariateSpline._from_tck(tck, self.ext)
+        return UnivariateSpline._from_tck(tck, ext=0)
```
Comment (ev-br) [inline scipy/interpolate/fitpack2.py]: This is not right. First, `ext=0` means "extrapolate", not "zeros". Second, the extrapolation mode should be set to `zeros` if and only if the original extrapolation mode was `const`.

**Review by ev-br — CHANGES_REQUESTED** [review]
Also needs a test, which can be adapted from the original bug report.

**Comment by Bharat123rox** [comment]
@ev-br where should I add tests for this? In `test_fitpack2.py` or `test_fitpack.py`? Also, since arguments are evaluated from `splder()` and then changed, I'm still unclear on which method this test should be added.

**Comment by ev-br** [comment]
test_fitpack2::TestUnivariateSpline looks like a right place for this test. Please also mention it's a regression test for gh-10195 in a comment

**Comment by ev-br** [comment]
Superseded by https://github.com/scipy/scipy/pull/10310

**Pull request closed without merging**